doc/design/parallel_do.md

```diff
@@ -0,0 +1,161 @@
+# Design Doc: Parallel_Do in PaddlePaddle
+
+In PaddlePaddle, we use parallel_do primitive to represent multithread data parallel processing.
+
+## Design overview
+
+The definition of a parallel_do op looks like the following
+
+```c++
+AddInput(kInputs, "Inputs needed to be split onto different devices").AsDuplicable();
+AddInput(kParameters, "Parameters are duplicated over different devices")
+    .AsDuplicable();
+AddInput(kPlaces, "Devices used for parallel processing");
+AddOutput(kOutputs, "Outputs needed to be merged from different devices").AsDuplicable();
+AddOutput(kParallelScopes,
```

Comment:
`kParallelScopes` seems to indicate that there are multiple scopes, but the description says `Container`, which is a single container:
1. does container mean scope?
1. is there a single scope or multiple scopes?

Reply:
1. yes
2. one scope for each device

Reply:
Maybe change "container" to "scope" and make "one scope for each device" clear? :)